k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolderPusher.java

```diff
@@ -253,6 +253,8 @@ private void reacquireWakeLockAndCleanUp() {
             } catch (Exception me) {
                 Log.e(LOG_TAG, "Got exception while closing for exception for " + getLogId(), me);
             }
+
+            connection = null;
```

Comment:
well, not a concrete case, but it's called first thing in a `catch(Exception)` block so I'd say any assumptions made on the state are unstable at best.